### PR TITLE
Dst 249  from spectrum1d

### PR DESCRIPTION
**Pull request by vegard-solum-4ss**
Construct DirectionalSpectrum from 1-D spectrum, spreading function and peak direction: https://4insight.atlassian.net/jira/software/c/projects/DST/boards/10?selectedIssue=DST-249

### CODE REVIEW

File: src/waveresponse/_core.py [hunk 1]
```diff
@@ -884,6 +884,89 @@ def from_grid(cls, grid):
...
+            'counterclockwise' is assumed.
+        waves_coming_from : bool
+            If waves are 'coming from' the given directions. If ``False``, 'going towards'
+            convention is assumed.
+        """
+        freq = np.asarray_chkfinite(freq)
+        dirs = np.asarray_chkfinite(dirs)
+        spectrum1d = np.asarray_chkfinite(spectrum1d)
+        vals = (
+            np.repeat(spectrum1d, len(dirs))
```
Comment (ace-e4s) [inline src/waveresponse/_core.py:945]: consider using `np.tile` instead of `np.repeat`

Comment (ace-e4s) [reply src/waveresponse/_core.py:945]: Then, reshape would be obsolete.

File: src/waveresponse/_core.py [hunk 2]
```diff
@@ -884,6 +884,89 @@ def from_grid(cls, grid):
...
+
+        if degrees:
+            period = 360.0
+        else:
+            period = 2.0 * np.pi
+
+        for (idx_f, idx_d), val_i in np.ndenumerate(vals):
+            f_i = freq[idx_f]
+            d_i = (dirs[idx_d] - dirp) % period
+            vals[idx_f, idx_d] = spread_fun(f_i, d_i) * val_i
```
Comment (ace-e4s) [inline src/waveresponse/_core.py:958]:
```suggestion
        dirs_spread = (dirs - dirp) % period
        for (idx_f, idx_d), val_i in np.ndenumerate(vals):
            vals[idx_f, idx_d] = spread_fun(freq[idx_f], dirs_spread[idx_d]) * val_i
```
Small vectorization... Effect is probably minimal. More readible?

**Review by ace-e4s — CHANGES_REQUESTED** [review]
Looks very good. Lets talk if you disagree with my suggestions.

File: src/waveresponse/_core.py (hunk 1 above)
Comment (vegard-solum-4ss) [reply src/waveresponse/_core.py]: Fixed

**Review by ace-e4s — APPROVED** [review]
🥇

**Pull request merged**